Do not break if a day only contains a single line.

icemac opened the pull request:
Render a better error message if an entry cannot be parsed.

icemac committed:
Do not break if a day only contains a single line.

Render a better error message if an entry cannot be parsed.

sallner commented on gtimelog2tick.py:
```diff
@@ -265,6 +273,8 @@ def parse_timelog(
         task, text, task_id = parse_entry_message(config, entry.message)
         worklog = WorkLog(entry, text, task, task_id)
         if worklog.hours > 0:
+            if worklog.task_id is None:
+                raise DataError(worklog.text)
```
I see now, that we need to calculate the hours to handle the first entry (which has `worklog.hours == 0` but is only handled implicitly here).

icemac replied:
Yes this was also my problem: to be able to ignore entries with 0 minutes you have to parse the message to create a Worklog instance. Maybe the factoring is not great here: It could be an idea to move parsing the message into the Worklog class and do it when necessary.

Let me try this.

icemac replied:
I refactored it in https://github.com/minddistrict/gtimelog2tick/pull/11/commits/5c3d819c01066fa8a4ccb2109d3f9dd3cb924c79.

sallner reviewed (approved):
I have some suggestions from reading the code, but did not try it out by myself.

icemac committed:
Apply suggestions from code review

Co-authored-by: Steffen Allner <s.allner@minddistrict.com>

icemac committed:
Refactor evaluation of entry message to be lazy.

Now a worklog entry can be created and the hours of it accessed before parsing
the message into a task.
